### PR TITLE
sql/postgres: marshal HCL enum

### PR DESCRIPTION
**Pull request by masseelch**
This is the `Marshal` part of the postgres support for enums.

### CODE REVIEW

File: sql/postgres/sqlspec.go [hunk 1]
```diff
@@ -63,7 +63,30 @@ func UnmarshalSpec(data []byte, unmarshaler schemaspec.Unmarshaler, v interface{
 
 // MarshalSpec marshals v into an Atlas DDL document using a schemaspec.Marshaler.
 func MarshalSpec(v interface{}, marshaler schemaspec.Marshaler) ([]byte, error) {
-	return specutil.Marshal(v, marshaler, schemaSpec)
+	d := &doc{}
```
Comment (yonidavidson) [inline sql/postgres/sqlspec.go:66]: var d doc

File: sql/postgres/sqlspec.go [hunk 2]
```diff
@@ -63,7 +63,30 @@ func UnmarshalSpec(data []byte, unmarshaler schemaspec.Unmarshaler, v interface{
...
+	switch s := v.(type) {
+	case *schema.Schema:
+		spec, tables, enums, err := schemaSpec(s)
```
Comment (yonidavidson) [inline sql/postgres/sqlspec.go:69]: I am not a fan of this API extending the number of returned values.
maybe we should return (struct,err) ?

File: sql/postgres/sqlspec.go [hunk 3]
```diff
@@ -216,9 +239,32 @@ func enumName(ref *schemaspec.Type) (string, error) {
...
+func enumRef(n string) *schemaspec.Ref {
+	return &schemaspec.Ref{
+		V: "$enum." + n,
+	}
+}
+
 // schemaSpec converts from a concrete Postgres schema to Atlas specification.
-func schemaSpec(schem *schema.Schema) (*sqlspec.Schema, []*sqlspec.Table, error) {
-	return specutil.FromSchema(schem, tableSpec)
+func schemaSpec(schem *schema.Schema) (*sqlspec.Schema, []*sqlspec.Table, []*Enum, error) {
```
Comment (yonidavidson) [inline sql/postgres/sqlspec.go:250]: https://github.com/ariga/atlas/pull/447/files#r788467466

File: sql/postgres/sqlspec.go (hunk 1 above)
Comment (masseelch) [reply sql/postgres/sqlspec.go:66]: `var d doc` <- not a pointer

File: sql/postgres/sqlspec.go (hunk 2 above)
Comment (masseelch) [reply sql/postgres/sqlspec.go:69]: schemaSpec is unimported and therefore I think this is okay. If it had been an exported function, I'd agree with you. Let's see what @a8m thinks :-)

File: sql/postgres/sqlspec.go (hunk 3 above)
Comment (a8m) [reply sql/postgres/sqlspec.go:250]: I'd return a doc instead.

**Review by a8m — APPROVED** [review]
Looks good, besides one comment. Returning more than 2 values from a function is not that common in this project.

**Review by rotemtam — APPROVED** [review]

**Pull request merged**